clients/pkg/logentry/stages/metrics.go

```diff
@@ -142,6 +142,19 @@ type metricStage struct {
 	metrics map[string]prometheus.Collector
 }
 
+func (m *metricStage) Run(in chan Entry) chan Entry {
+	out := make(chan Entry)
+	go func() {
+		defer close(out)
+
+		for e := range in {
+			m.Process(e.Labels, e.Extracted, &e.Timestamp, &e.Line)
+			out <- e
+		}
+	}()
+	return out
+}
```

Comment:
why was this change necessary? just to mimic what `toStage` was giving us before?

Reply:
Yes. The metrics stage needs a cleanup function, and `toStage` doesn't add one. Maybe we could avoid the repetition by having another function called `toStageWithCleanup`?

Reply:
I think this way is fine. Even with `toStageWithCleanup` we would still require the addition of `Cleanup` noop functions on all stages except the metric stage.